examples/BCDR/BackupFromStorageQueue.cs

```diff
@@ -0,0 +1,172 @@
+                    {
+                        JObject eventObject = JObject.Parse(decodedMessage);
+                        string topic = eventObject.Property("topic")?.Value.ToString();
+                        if (topic != null && topic == Environment.GetEnvironmentVariable("PrimaryStoreResourceId"))
+                        {
+                            JObject dataObject = JObject.Parse(eventObject.Property("data")?.Value.ToString());
+                            string key = dataObject["key"].ToString();
+                            string label = dataObject["label"]?.ToString();
+                            updatedKeyLabels.Add(new KeyLabel(key, label));
+                        }
+                    }
+                    catch (JsonReaderException)
+                    {
+                        // If its not a valid JSON, ignore the message 
+                        log.LogInformation($"Queue message in invalid JSON format will be ignored.\nMessage: {decodedMessage}");
+                    }
+                }
+                catch (FormatException)
+                {
+                    // If its not in valid base64 format, ignore the message 
+                    log.LogInformation($"Queue message in invalid Base64 format will be ignored.\nMessage: {message.MessageText}");
+                }
+            }
+            return updatedKeyLabels;
+        }
+
+        private static void BackupKeyValues(HashSet<KeyLabel> updatedKeyLabels, ILogger log)
+        {
+            // Connect to Azure App Configuration stores
+            ConfigurationClient primaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("PrimaryStoreEndpoint")), new ManagedIdentityCredential());
+            ConfigurationClient secondaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("SecondaryStoreEndpoint")), new ManagedIdentityCredential());
+
+            // Read all settings from Primary Store and update secondary store
+            foreach (ConfigurationSetting setting in primaryAppConfigClient.GetConfigurationSettings(new SettingSelector()))
+            {
+                KeyLabel primaryStoreKeyLabel = new KeyLabel(setting.Key, setting.Label);
+                if(updatedKeyLabels.Contains(primaryStoreKeyLabel))
+                {
+                    // current setting retrieved from primary store needs to be updated in secondary store
+                    secondaryAppConfigClient.SetConfigurationSetting(setting);
+                    log.LogInformation($"Successfully updated key: {setting.Key} label: {setting.Label}");
+                    updatedKeyLabels.Remove(primaryStoreKeyLabel);
+                }
+            }
+
+            // Delete key labels still present in updatedKeyLabels from secondary store
+            foreach (var keyLabel in updatedKeyLabels)
+            {
+                secondaryAppConfigClient.DeleteConfigurationSetting(keyLabel.Key, keyLabel.Label);
```

Comment:
Can you please make sure this will not throw for things like 404?

Reply:
This will throw RequestFailedException for anything other than 200 (delete successful) and 204 (Key not found, nothing to delete). RequestFailedException is being handled appropriately.

Reply:
If this throws a `RequestFailedException` with `404`, then the current implementation would get stuck in an infinite processing cycle since subsequent attempts to process these messages would also get the same error. Instead, we can simply ignore such an error here and proceed as normal, since the key-value that we're trying to delete is already in the desired state (non-existent).

Reply:
If the key value is already in desired state (non-existent), it will not throw 404 with exception. Instead, it returns 204, which is considered a success response. If we still get 404 somehow, it means something is actually wrong, and we abort backing up any key values. In fact, if there is any exception thrown by our SDK, we abort the current function execution and rely on the next triggered execution to process the entire backlog of event messages.

Reply:
Okay, I just checked the SDK code and it does not seem to throw `404`. It dies however throw `409` if the setting is read-only. Have you tried replicating a read-only setting?

Reply:
Yes, if users have somehow locked a setting in their secondary store and update/delete the same setting in primary store, we will keep hitting this exception and crashing the function on every run. Since we currently do not support backing up locking/unlocking of keys, the only way to hit this exception is if users manually lock a key in secondary store.

Reply:
Here's a scenario. A user edits a setting and locks the key. When we receive the event for setting being updated, we pull the read only setting and write to the secondary store. Now if the user unlocks the setting and updates it, we will hit this issue and the backup would get stuck.

Reply:
Looking at the SDK code, we dont actually set key as readonly with the SetConfigurationSettingAsync(ConfigurationSetting) call. There is a separate function for creating/updating read only keys called SetReadOnlyAsync (https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs#L887 ).

We can handle this in two ways:
1. We don't bother locking keys in secondary store at all.
2. We check if a setting is readonly, and call the appropriate SetReadOnlyAsync API for locked keys.

The problem with second approach is - we don't receive any events if key is simply locked/unlocked in primary store. So the only way to backup lock/unlock changes is if they are accompanied with update/delete changes. In my opinion, it would just create an unpredictable inconsistency between the primary and secondary stores (and also make the sample code snippet more complex).

Reply:
So in the current implementation, we do support backup for read-only keys, but the read-only flag would not be persisted in the secondary store. This should be fine for now. When we do start firing events for lock/unlock changes, we can revisit this and see if we can also backup the read-only flag.

Reply:
Sounds good.

Reply:
Let's make it explicit. Avani, in the instructions, please call out subscribing only to the "modified" and "deleted" events, so we don't need to worry about whether/when we add the "lock-modified" event in the future.

Reply:
Added explicit instructions in the tutorial to subscribe to only KV modified and deleted events.